src/mbed_tools/build/_internal/config/assemble_build_config.py

```diff
@@ -84,13 +84,7 @@ def _get_app_filter_labels(mbed_app_data: dict, config: Config) -> None:
 
 
 def _get_file_filter_overrides(mbed_app_data: dict) -> dict:
-    return {
-        "overrides": [
-            override
-            for override in mbed_app_data.get("overrides", [])
-            if override.modifier or override.name == "requires"
-        ]
-    }
+    return {"overrides": [override for override in mbed_app_data.get("overrides", []) if override.modifier]}
```

Comment:
It may be worth mentioning in the commit message for "config: Remove support for target.requires" that this partially reverts 6a99b43bf11f8bebf52b6f3957a2653f8da03258

Reply:
Mentioned.